src/main/java/Du/TaskList.java

```diff
@@ -80,4 +80,29 @@ public void find_finished_task(int order) {
         complete_task.finish_task();
     }
 
+
+    /**
+     *
+     * @param substring
+     * @return
+     */
+    public ArrayList<Task> search(String substring) {
+        ArrayList<Task> have_substring = new ArrayList<>();
+        for (Task t : this.getList_of_tasks()) {
+            if (t.getName().contains(substring)) {
+                have_substring.add(t);
+            }
+        }
+        return have_substring;
+    }
+
+    public void print(ArrayList<Task> arraylist) {
+        System.out.println("____________________________________________________________\n" +
```

Comment:
This is what I did at first as well. It turns out that a more recommended approach is to set a global final String variable that contains this line. It could potential make the code cleaner.